packages/babel-plugin-proposal-decorators/test/fixtures/2023-11-duplicated-keys--to-es2015/computed-keys-same-value/exec.js

```diff
@@ -1,29 +1,30 @@
-expect(() => {
-  var i = 0;
-  var j = 0;
+var i = 0;
+var j = 0;
 
-  function getKeyI() {
-    return (i++).toString();
-  }
-  function getKeyJ() {
-    return (j++).toString();
-  }
+function getKeyI() {
+  return (i++).toString();
+}
+function getKeyJ() {
+  return (j++).toString();
+}
 
-  let elements = [];
+let elements = [];
 
-  function dec(fn, context) {
-    elements.push({ fn, context });
-  }
+function dec(fn, context) {
+  elements.push(fn);
+}
 
-  class Foo {
-    @dec
-    [getKeyI()]() {
-      return 1;
-    }
+class Foo {
+  @dec
+  [getKeyI()]() {
+    return 1;
+  }
 
-    @dec
-    [getKeyJ()]() {
-      return 2;
-    }
+  @dec
+  [getKeyJ()]() {
+    return 2;
   }
-}).toThrow("Attempted to decorate a public method/accessor that has the same name as a previously decorated public method/accessor. This is not currently supported by the decorators plugin. Property name was: 0")
+}
+
+expect(elements[0]()).toEqual(2);
```

Comment:
Is it correct that this returns `2` and not `1`?

Reply:
It seems so. I'm not sure if the spec supports such usage. ts behaves the same as after this PR.
The size of this check is ~10% of the helper's. 🤔

Reply:
@pzuraq Do you have a couple examples of what is the expected behavior when decorating class fields & methods that have the same name?

- two prototype methods
- static method and prototype method
- instance field and prototype method
- two instance fields

Reply:
Decorators apply to each individual element independently, and then the element is applied to the class definition in the same order and with the same timing and semantics as if they were not decorated. So:

> two prototype methods

1. Decorators are applied to the first method
2. First method is assigned to the class prototype
3. Decorators are applied to the second method
4. Second method is assigned to the class prototype, overwriting the first method

> static method and prototype method
1. Decorators are applied to the static method
2. Static method is assigned to the class
3. Decorators are applied to the prototype method
4. Prototype method is assigned to the class prototype

>  instance field and prototype method
1. Decorators are applied to the prototype method
2. Prototype method is assigned to the class prototype
3. Decorators are applied to the instance field
4. On initialization, the instance field is `[Define]`d on the instance, shadowing the prototype method 

> two instance fields
1. Decorators are applied to instance field 1
2. Decorators are applied to instavce field 2
3. Instance field 1 is initialized and defined on the instance
4. Instance field 2 is initialized and defined on the instance, overwriting instance field 1

In every case, the decoration phase is not altered at all by the existence of another element with the same name. There is no side-channeled information, etc. 

When they are the same `static`, the only case that doesn't end up with the element being overwritten/shadowed is getter and setter with the same name, which will coalesce into a normal getter setter pair.

Reply:
Thank you!

@liuxingbaoyu At least for the "two methods" case we are not implementing the proposal properly -- let's keep the error, but we can shorten the message for example to `"Decorating two elements with the same name (" + name + ") is not supported yet"`

Reply:
I don't love the size of this, but I'll keep it in this PR.
I refactored this check, but it still accounts for ~7% of the entire helper, also after gzip.
But luckily we reduced some other sizes inspired by https://github.com/babel/babel/pull/16261.